jax/_src/numpy/util.py

```diff
@@ -464,12 +469,9 @@ def size(a: ArrayLike | SupportsSize | SupportsShape, axis: int | None = None) -
     >>> y.size
     6
   """
-  if (axis is None and hasattr(a, "size")) or (axis is not None and hasattr(a, "shape")):
-    # NumPy dispatches to a.size/a.shape if available.
-    return np.size(a, axis=axis)  # type: ignore[arg-type]
-  # Deprecation warning added 2025-2-20.
+  if axis is None and hasattr(a, "size"):
+    return a.size
   check_arraylike("size", a, emit_warning=True)
```

Comment:
Please put this above the previous check. We should require array-like input regardless of whether `axis` is None.